### PR TITLE
#6115: add a q, k, v heads creation api that can take a split q and k…

### PR DESCRIPTION
**Pull request by sjameelTT**
…v tensor with differing sequence lengths

- Adds cross-attention support
- Also a step for addressing #7519
- TODO: add batch > rows support

### CODE REVIEW

File: tests/tt_eager/python_api_testing/unit_testing/misc/test_create_qkv_heads.py [hunk 1]
```diff
@@ -165,3 +165,183 @@ def test_nlp_create_qkv_heads_test(
...
+        pcc = 0.99
+    elif (
+        dtype == ttl.tensor.DataType.FLOAT32 and transpose_k
+    ):  # conversion from fp32 to tf32 when unpack writes to register for compute will decrease pcc in the transpose case
+        pcc = 0.9999999
+    else:
+        pcc = 1.0
+
+    print(pyt_got_back_rm_k)
+    print(ref_k)
```
Comment (TT-BrianLiu) [inline tests/tt_eager/python_api_testing/unit_testing/misc/test_create_qkv_heads.py:292]: Remove

File: tt_eager/tt_dnn/op_library/nlp_tms/kernels/dataflow/reader_create_qkv_heads_sharded_separate.cpp [hunk 2]
```diff
@@ -0,0 +1,122 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdint.h>
+#include "dataflow_api.h"
+#include "debug/dprint.h"
```
Comment (TT-BrianLiu) [inline tt_eager/tt_dnn/op_library/nlp_tms/kernels/dataflow/reader_create_qkv_heads_sharded_separate.cpp:7]: Remove

**Review by TT-BrianLiu — APPROVED** [review]

File: tests/tt_eager/python_api_testing/unit_testing/misc/test_create_qkv_heads.py (hunk 1 above)
Comment (sjameelTT) [reply tests/tt_eager/python_api_testing/unit_testing/misc/test_create_qkv_heads.py:292]: Nice catch, thanks, done

File: tt_eager/tt_dnn/op_library/nlp_tms/kernels/dataflow/reader_create_qkv_heads_sharded_separate.cpp (hunk 2 above)
Comment (sjameelTT) [reply tt_eager/tt_dnn/op_library/nlp_tms/kernels/dataflow/reader_create_qkv_heads_sharded_separate.cpp:7]: Done

**Pull request merged**